### PR TITLE
Fix channel payouts mappings and add missing tests

### PR DESCRIPTION
**Pull request by Lezek123**
Addresses https://github.com/Joystream/joystream/issues/4579 and adds test coverage

### CODE REVIEW

File: query-node/mappings/src/content/channel.ts [hunk 1]
```diff
@@ -501,7 +501,7 @@ export async function content_ChannelRewardClaimedAndWithdrawn({
   event,
 }: EventContext & StoreContext): Promise<void> {
   // load event data
-  const [owner, channelId, amount, accountId] = new Content.ChannelRewardClaimedAndWithdrawnEvent(event).params
+  const [owner, channelId, amount, destination] = new Content.ChannelRewardClaimedAndWithdrawnEvent(event).params
```
Comment (dobertRowneySr) [inline query-node/mappings/src/content/channel.ts]: maybe replace `amount` with `withdrawnAmount` since it's easier to understand

**Review by dobertRowneySr — CHANGES_REQUESTED** [review]
it's ok, I just left a cosmetic suggestion.

 I just want to mention 2 things:
1. we are not having any tests for curator channel in which `claim_and_withdraw` would leave the council budget unaffected theoretically. 
2. we are not making any checks to ensure proper `api.query.council.budget` variations

I am not sure if they should be implemented here or they can be checked in other way (QA)

File: query-node/mappings/src/content/channel.ts (hunk 1 above)
Comment (Lezek123) [reply query-node/mappings/src/content/channel.ts]: Done in https://github.com/Joystream/joystream/pull/4594/commits/d7013f7523758e1d1211056e968d91fc2544002e

**Comment by Lezek123** [comment]
> we are not having any tests for curator channel in which claim_and_withdraw would leave the council budget unaffected theoretically.

We have unit tests in the runtime, but we don't have integration tests. We don't really have integration tests for curator channels in general, which is quite unfortunate.

> we are not making any checks to ensure proper api.query.council.budget variations

Added some basic checks in https://github.com/Joystream/joystream/pull/4594/commits/1320d219cc9dacba79ab5087f39867511b94768f

**Review by dobertRowneySr — APPROVED** [review]
LGTM, merging

**Pull request merged**